pallets/handles/src/tests/handle_retirements_tests.rs

```diff
@@ -53,6 +54,64 @@ fn claim_and_retire_handle_happy_path() {
 	});
 }
 
+#[test]
+fn test_handle_early_retirement_fails() {
+	new_test_ext().execute_with(|| {
+		let base_handle_str = "test1";
+
+		let alice = sr25519::Pair::from_seed(&[0; 32]);
+		let expiration = 50;
+		let (payload, proof) =
+			get_signed_claims_payload(&alice, base_handle_str.as_bytes().to_vec(), expiration);
+		assert_ok!(Handles::claim_handle(
+			RuntimeOrigin::signed(alice.public().into()),
+			alice.public().into(),
+			proof.clone(),
+			payload.clone()
+		));
+		let msa_id = MessageSourceId::decode(&mut &alice.public().encode()[..]).unwrap();
+		let handle = Handles::get_handle_for_msa(msa_id);
+		assert!(handle.is_some());
+
+		// Compose the full display handle from the base handle, "." delimeter and suffix
+		let handle_result = handle.unwrap();
+		let suffix: HandleSuffix = handle_result.suffix;
+		let mut full_handle_vec: Vec<u8> = vec![];
+		full_handle_vec.extend(base_handle_str.as_bytes());
+		full_handle_vec.extend(b"."); // The delimeter
+		let mut buff = [0u8; SUFFIX_MAX_DIGITS];
+		full_handle_vec.extend(suffix.numtoa(10, &mut buff)); // Use base 10
+
+		// Fail to retire due to mortality
+		let call_retire_handle: &<Test as frame_system::Config>::RuntimeCall =
+			&RuntimeCall::Handles(HandlesCall::retire_handle {});
+		let info = DispatchInfo::default();
+		let len = 0_usize;
+		let early_retire_result = HandlesSignedExtension::<Test>::new().validate(
+			&alice.public().into(),
+			call_retire_handle,
+			&info,
+			len,
+		);
+		assert!(early_retire_result.is_err());
```

Comment:
Do we need to make sure that we are getting the expected here we want here?

Reply:
We are safe from the point of view of that this is a direction function call so it is expected to return error, but I will see if there is a way to do some comparison